clang/lib/Sema/CheckedCAnalysesPrepass.cpp

```diff
@@ -70,16 +82,17 @@ class PrepassHelper : public RecursiveASTVisitor<PrepassHelper> {
       return T->getAsRecordDecl();
     }
 
-    // AddBoundsSiblingField adds FieldWithBounds to the set of fields in
-    // whose declared bounds F occurs.
-    void AddBoundsSiblingField(const FieldDecl *F) {
-      auto It = Info.BoundsSiblingFields.find(F);
-      if (It != Info.BoundsSiblingFields.end())
-        It->second.insert(FieldWithBounds);
+    // AddPtrWithBounds adds PtrWithBounds to the set of variables in whose
+    // bounds Var occurs.
+    template<class M, class P, class V>
+    void AddPtrWithBounds(M &Map, const P *PtrWithBounds, const V *Var) {
```

Comment:
Do any calls to `AddPtrWithBounds` use different types for `PtrWithBounds` and `Var`?

Reply:
I am not sure I understood your question. All calls to `AddPtrWithBounds` use different types for `PtrWithBounds` and `Var`.

Reply:
I thought that, for any given call to `AddPtrWithBounds`, `PtrWithBounds` and `Var` would either both be of type `VarDecl *` or both be of type `FieldDecl *`. My question is, do we need separate template type arguments `class P` and `class V` for `AddPtrWithBounds`?